plugins/RedundantLinkManager/Link.cs

```diff
@@ -0,0 +1,330 @@
+﻿using MissionPlanner;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace RedundantLinkManager
+{
+    /// <summary>
+    /// Options/Settings and MAVLinkInterface for a single link. When the settings are changed, the MAVLinkInterface
+    /// is disposed so that it can be reconnected by the autoconnect thread.
+    /// </summary>
+    public class Link : ICloneable, IDisposable, IEquatable<Link>
+    {
+        private bool _enabled;
+        public bool Enabled
+        {
+            get { return _enabled; }
+            set
+            {
+                if (_enabled == value) return;
+                _enabled = value;
+                Dispose();
+            }
+        }
+
+        private string _type;
+        public string Type
+        {
+            get { return _type; }
+            set
+            {
+                if (_type == value) return;
+                _type = value;
+                Dispose();

```

Comment:
why dispose here ?

Reply:
These Link objects are bound to the DataGridView rows in the manager UI. If any of the settings change that would require the link to be closed and reopened, I need to dispose.